rasa/core/domain.py

```diff
@@ -735,38 +736,49 @@ def persist(self, filename: Text) -> None:
         domain_data = self.as_dict()
         utils.dump_obj_as_yaml_to_file(filename, domain_data)
 
-    def persist_clean(self, filename: Text) -> None:
-        """Write domain to a file.
-
-         Strips redundant keys with default values."""
-
-        data = self.as_dict()
-
-        for idx, intent_info in enumerate(data["intents"]):
+    @staticmethod
+    def _clean_domain(domain_data: Dict[Text, Any]) -> Dict[Text, Any]:
+        for idx, intent_info in enumerate(domain_data["intents"]):
             for name, intent in intent_info.items():
                 if intent.get("use_entities"):
-                    data["intents"][idx] = name
+                    domain_data["intents"][idx] = name
 
-        for slot in data["slots"].values():
+        for slot in domain_data["slots"].values():
             if slot["initial_value"] is None:
                 del slot["initial_value"]
             if slot["auto_fill"]:
                 del slot["auto_fill"]
             if slot["type"].startswith("rasa.core.slots"):
                 slot["type"] = Slot.resolve_by_type(slot["type"]).type_name
 
-        if data["config"]["store_entities_as_slots"]:
-            del data["config"]["store_entities_as_slots"]
+        if domain_data["config"]["store_entities_as_slots"]:
+            del domain_data["config"]["store_entities_as_slots"]
 
         # clean empty keys
-        data = {k: v for k, v in data.items() if v != {} and v != [] and v is not None}
+        return {
+            k: v
+            for k, v in domain_data.items()
+            if v != {} and v != [] and v is not None
+        }
+
+    def persist_clean(self, filename: Text) -> None:
+        """Write domain to a file.
+
+         Strips redundant keys with default values."""
 
-        utils.dump_obj_as_yaml_to_file(filename, data)
+        domain_data = self.as_dict()
+        cleaned_domain_data = self._clean_domain(domain_data)
+        utils.dump_obj_as_yaml_to_file(filename, cleaned_domain_data)
 
     def as_yaml(self):
         domain_data = self.as_dict()
         return utils.dump_obj_as_yaml_to_string(domain_data)
 
+    def as_yaml_clean(self):
```

Comment:
should this maybe just be an argument to `def as_yaml(self, clean_before_dump=False)`